strax/context.py

```diff
@@ -948,6 +926,70 @@ def concat_loader(*args, **kwargs):
             loader_plugins=loader_plugins,
             savers=savers,
             targets=strax.to_str_tuple(final_plugin))
+    
+    def _add_saver(self, savers, run_id, d_to_save, target_plugin, _is_superrun, loading_this_data):
+        key = strax.DataKey(run_id, d_to_save, target_plugin.lineage)
```

Comment:
Could you make a docstring / type hints since you also confused yourself with the meaning of target, target_i and target plugin:
```python
key = strax.DataKey(run_id, d_to_save)
```

Reply:
I wonder why this is not covered by the tests 🤔 , do you see an easy way we would fail here?

Reply:
whoops, I see I was actually confused, but I think target_plugin is sometimes passed as a string

Reply:
> Could you make a docstring / type hints since you also confused yourself with the meaning of target, target_i and target plugin:

Sure, but I cannot follow for your other two comments. 

> I wonder why this is not covered by the tests

According to coveralls it is is not it? 

> whoops, I see I was actually confused, but I think target_plugin is sometimes passed as a string

What do you mean? At least not during testing otherwise the function would fail.

Reply:
Hmm okay, yeah as I said I was confused, looking a bit closer I must have been messing up with `target_i` sorry for the jitter ( `get_components` must be one of the more confusing functions in my defense :joy:)

Reply:
you might by the way get rid of the `run_id` argument in  `_add_saver` by using `plugin.run_id` instead if you want to reduce the number of required arguments

Reply:
> get_components must be one of the more confusing functions in my defense

Yes, absolutely agree on that :D